Project Pathfinder/Assets/Scripts/GenerateMaze.cs

```diff
@@ -0,0 +1,141 @@
+        Stack<CellPosition> positionStack = new Stack<CellPosition>(); // Holds maze cell position data
+        System.Random randomNum           = new System.Random();       // Random number generator
+        CellPosition initialCell          = new CellPosition{ _x = randomNum.Next(0, mazeWidth), _y = randomNum.Next(0, mazeHeight) },
+                                                                       // Starting cell for path generation
+                     currentCell          = new CellPosition(),        // Position of the current cell being processed
+                     neighborCell         = new CellPosition();        // Position of the selected neighbor of the current cell
+        List<CellNeighbor> cellNeighbors  = new List<CellNeighbor>();  // List of unvisited neighbors to the current cell  
+        CellNeighbor randomNeighbour      = new CellNeighbor();        // Randomly selected neighbor from the list of current cell neighbors
+        int randomIndex;                                               // Index used to randomly select a cell neighbor
+        
+        // Process the initial cell as part of the maze path
+        mazeData[initialCell._x, initialCell._y] |= WallStatus.VISITED;
+        positionStack.Push(initialCell);
+
+        // Generate maze paths until all maze cells have been visited
+        while(positionStack.Count > 0){
+            currentCell   = positionStack.Pop();
+            cellNeighbors = GetUnvisitedNeighbours(currentCell, mazeData, mazeWidth, mazeHeight);
+
+            // If the current cell has an unvisited neighbor, add the neighbor cell to the current maze path
+            if(cellNeighbors.Count > 0){
+                // Add the current cell to the path history
+                positionStack.Push(currentCell);
+
+                // Randomly select an unvisited neighbor
+                randomIndex     = randomNum.Next(0, cellNeighbors.Count);
+                randomNeighbour = cellNeighbors[randomIndex];
+                neighborCell    = randomNeighbour._position;
+
+                // Delete the maze wall separating the current cell from its neighbor
+                mazeData[currentCell._x, currentCell._y] &= ~randomNeighbour._sharedWall;
+                mazeData[neighborCell._x, neighborCell._y] &= ~GetOppositeWall(randomNeighbour._sharedWall);
+
+                // Set the neighbor cell as the new current cell
+                mazeData[neighborCell._x, neighborCell._y] |= WallStatus.VISITED;
+                positionStack.Push(neighborCell);
+            }
+        }
+        return mazeData;
+    }
+
+    // Get a list of the current cell's unvisited neighbors
+    private static List<CellNeighbor> GetUnvisitedNeighbours(CellPosition currentCell, WallStatus[,] mazeData, int mazeWidth, int mazeHeight){
+        List<CellNeighbor> cellNeighbors = new List<CellNeighbor>(); // List of unvisited neighbors to the current cell
```

Comment:
See previous comment about changing cellNeighbors to "unvisitedCellNeighbors"

Reply:
Done.
changed it to "unvisitedNeighbors" so it matches the name of the function